src/main/kotlin/com/sourcegraph/cody/CodyToolWindowContent.kt

```diff
@@ -28,9 +30,15 @@ class CodyToolWindowContent(private val project: Project) {
 
   private var codyOnboardingGuidancePanel: CodyOnboardingGuidancePanel? = null
   private val signInWithSourcegraphPanel = SignInWithSourcegraphPanel(project)
-  private val historyTree = JPanel()
+  private val historyTree = HistoryTree(::selectHistory, onDelete = {
+      println("on delete")
+  })
   private val tabbedPane = JBTabbedPane()
 
+  private fun selectHistory(state: ChatState) {
+    addChatSession(AgentChatSession.createFromState(project, state))
```

Comment:
How does that work if someone select the same chats back and forth multiple times?
Won't it duplicate existing sessions?
I would thought we need to do something like:

```
val session = AgentChatSession.getSessionByInternalId(state.id) ?: AgentChatSession.createFromState(project, state)
addChatSession(session)
```

(better name for `addChatSession ` would probably be `selectChatSession`, and I know I'm guilty of that bad name ;))

Reply:
Good catch! That fixed one of my problems.